### PR TITLE
feat: actions workflow

### LINKED ISSUE #177: The Prettier linter, main.yml GitHub Actions Flow and Pull Request UI
- There is a slight issue with #173 where the GitHub `prettier-lint-check.yml` action happens on `push and pull_request update` whereas `main.yml` runs on `push`.

I've separated this issue as I wanted more of the community to get involved in understanding GitHub Actions and I'd love some feedback on what you think.

As in CircleCI, Travis and similar, you will find most CI/CD pipelines to be somewhat procedural, with the exceptions of well crafted "parallelism" where all parallel jobs must have a valid exit code or the entire job fails. You can also specify otherwise, e.g. parallelise all the things but let the developer know which tests did fail. (Think more than one language/stack in a mono-repo.)

This does open a problem as even if it does find a Prettier fault, it will ❌ that action, however that has ran **after** it has gone ahead and ran `main.yml`.

This may just be me, but this UI suggests quite the opposite of what is happening here:

image

### Suggestion:
Suggestion is to:

- collaborate on how to order these actions, check if these `on: [push]` and `on: [push, request]` is correct
- work on how we ensure that Prettier failing is really clear in the GitHub UI.
  - Because the `main.yml` awesome gihub profiles generator will run in parallel of the prettier-lint-check, even if the JSON is not exactly "pretty", it will look like it's got the all clear from the GitHub action/
  - **This separated file may not be best.** as they run in parallel.

### PR DESCRIPTION
**Pull request by krishguptadev**
fixes #177

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -1,10 +1,39 @@
-on: [push]
+# @see https://github.com/EddieJaoudeCommunity/awesome-github-profiles/issues/173
```
Comment (krishguptadev) [inline .github/workflows/main.yml:1]:
```suggestion
# @see https://github.com/EddieHubCommunity/awesome-github-profiles/issues/173
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -15,3 +44,4 @@ jobs:
           columns: 3
           object-field-names: '[ "githubUsername", "name", "imageUrl", "issueNumber" ]'
           html-cell: '<td align="center"><p><a href="https://github.com/{{ githubUsername }}">{{ name }}</a></p><img src="{{ imageUrl }}" /><p><a href="https://github.com/EddieJaoudeCommunity/awesome-github-profiles/issues/{{ issueNumber }}">(💯 give your vote)</a></p></td>'
+
```
Comment (Panquesito7) [inline .github/workflows/main.yml:47]: There's an unneeded extra line here (the Git diff doesn't display the endline, meaning there are two lines).
```suggestion
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -1,10 +1,39 @@
...
+          dry: True
+          # This defaults to only "**/*.js", so we specify what we have as file extensions here.
+          prettier_options: --write **/*.{md,json,yml,yaml}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
   table:
     runs-on: ubuntu-latest
+    needs: prettier
     name: Update README from json data
-    if: github.repository_owner == 'EddieHubCommunity'
```
Comment (Panquesito7) [inline .github/workflows/main.yml:7]: I think this one should be kept, not sure, though. @schmelto, @eddiejaoude?

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -1,10 +1,39 @@
...
+    if: github.event === 'push'
```
Comment (krishguptadev) [inline .github/workflows/main.yml:36]:
```suggestion
    if: ${{ github.event === 'push' && github.repository_owner == 'EddieHubCommunity' }}
```

Comment (krishguptadev) [reply .github/workflows/main.yml:36]: @Panquesito7 do you think, this would be good?

Comment (Panquesito7) [reply .github/workflows/main.yml:36]: Awesome! 🚀

**Review by Panquesito7 — APPROVED** [review]
LGTM 🚀 Thank you for your contribution! 😄👍

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -1,10 +1,39 @@
...
+          # This defaults to only "**/*.js", so we specify what we have as file extensions here.
+          prettier_options: --write **/*.{md,json,yml,yaml}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
   table:
     runs-on: ubuntu-latest
+    needs: prettier
     name: Update README from json data
-    if: github.repository_owner == 'EddieHubCommunity'
+    if: ${{ github.event === 'push' && github.repository_owner == 'EddieHubCommunity' }}
```
Comment (eddiejaoude) [inline .github/workflows/main.yml:36]: Not sure if the extra check of `event === 'push'` is needed as the trigger is `push` and `pull_request`?

Comment (krishguptadev) [reply .github/workflows/main.yml:36]: We don't want to update the readme if it is a pull request though 😅

Comment (krishguptadev) [inline .github/workflows/main.yml:36]:
```suggestion
    if: ${{ github.event == 'push' && github.repository_owner == 'EddieHubCommunity' }}
```

File: .github/workflows/main.yml [hunk 6]
```diff
@@ -1,10 +1,39 @@
-on: [push]
+# @see https://github.com/EddieHubCommunity/awesome-github-profiles/issues/173
+name: Prettier Lint checker to find any syntax/spacing/best practices tweaks.
```
Comment (krishguptadev) [inline .github/workflows/main.yml:2]:
```suggestion
name: Main
```

File: .github/workflows/main.yml [hunk 7]
```diff
@@ -1,10 +1,39 @@
...
+
+# @see https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestbranchestags
+# "Push" is every commit to any branch, whilst "pull_request" includes update commits to PRs.
+on: [push, pull_request]
```
Comment (krishguptadev) [inline .github/workflows/main.yml:6]:
```suggestion
on: [push, pull_request, workflow_dispatch]
```

**Comment by nhcarrigan** [comment]
Heya @krishguptadev - we've updated the workflow to run the prettier lint action from the `package.json` file instead. As such, these changes should no longer be needed (the action no longer commits anything).

We're going to go ahead and close this PR, but we encourage you to make further contributions!

**Pull request closed without merging**